game-core/src/main/java/games/strategy/triplea/attachments/UnitAttachment.java

```diff
@@ -1752,19 +1753,45 @@ private void resetCanDieFromReachingMaxDamage() {
   }
 
   private void setIsSuicide(final String s) {
-    isSuicide = getBool(s);
+    setIsSuicide(getBool(s));
   }
 
   private void setIsSuicide(final Boolean s) {
-    isSuicide = s;
+    setIsSuicideOnAttack(s);
+    setIsSuicideOnDefense(s && !Properties.getDefendingSuicideAndMunitionUnitsDoNotFire(getData()));
+    setIsFirstStrike(s);
+  }
+
+  private void setIsSuicideOnAttack(final String s) {
```

Comment:
Should the 's' convention be carried forward? Would it be more clear if the parameter were simply spelled out?

Reply:
I'd rather keep consistency for now. Easier to just update them all at once. Though honestly most of these are so simple that the parameter name isn't a huge deal.

Reply:
I'm not entirely sure it's consistent, I think there is argument for it being copy/pasta. Otherwise, we're left with just a single letter variable name representing the type, 'String'. Or does the 's' represent 'suicide', or 'isSuicide'? Having to decode what a variable single character represents is a small burden, but a burden nonetheless, that makes it harder to grok/read code.

Reply:
> "Easier to just update them all at once."

I have trouble agreeing as there are many more instances 'to fix' now. It's not clear they would all get the same fix.

Reply:
Actually, backing up a bit, the private setter is really odd. Perhaps would it be easier if we just inlined this? Then when invoked via method reference, it would imply that we have more going on then a simple setter. Someone reading the code would be no worse off seeing any of the inconsistent cases, either they'd see the simple setter inline, or they would see a method reference that is either more complex or is a simple setter (which we could later come back to clean up).

Reply:
I believe the double setters (1 string, 1 actual type) has to do with the XML parsing for attachments. The attachment classes in general are very rough and primarily there for to hold the values from the map XML parsing.

Reply:
> I believe the double setters (1 string, 1 actual type) has to do with the XML parsing for attachments. The attachment classes in general are very rough and primarily there for to hold the values from the map XML parsing.

I believe I understood that; My point is the indirection here does not seem useful, inlining these methods would reduce the indirection.

Re: variable name, most of the examples have been touched/added here and the new setter methods could have different variable names altogether. I don't really trust that we ever will come back and fix them.

Reply:
I think you're missing the big picture here. This entire class and really all the attachment classes follow this pattern of methods. Now whether that is an ideal pattern or not we can discuss but that is really a much larger issue than this PR and I would prefer to have a separate issue for it.

Reply:
I don't think I am necessarily:
- At 4300+ lines, this class is a monster. At such scale, it's arguable there are any patterns, it's a task just to grok it to see if anything is consistent or what it even contains.

This is a counter-example to the 'pattern':

```
        .put(
            "transportCost",
            MutableProperty.ofMapper(
                DefaultAttachment::getInt,
                this::setTransportCost,
                this::getTransportCost,
                () -> -1))
```


To help avoid unnecessary indirection I'm suggesting to follow the example of the counter-example and inline the setter/getters if they are simple.

Second, the class violates our preferred method ordering with arbitrarily located getter/setters that ought to be below where they are used.

Third, if not explicitly listed, we have a preference to spell out variables and not use single letter variables. I'm not positive if the 's' makes sense or is the right prefix, and seeing `s && ...` is not as easy to understand as it could be to understand.

I can understand pushback on the first and second points, though I don't think either of us should make that the hill we choose to die on, the third point seems far less reasonable to ignore as it's more about the care of this update as it is adding new code that should really not be adding at least that element of technical debt.

Reply:
I'd actually argue the larger the class the more consistency matters as its harder to understand if you see something being done multiple different ways. 

That being said, I misunderstood what you were suggesting about in lining. I'm open to using one of the helper mappers to reduce which setter/getter/reseter are needed. Updated this.